### PR TITLE
Optional kernels

### PR DESCRIPTION
**Pull request by rleander**
try-out

### CODE REVIEW

File: imod_coupler/drivers/ribametamod/config.py [hunk 1]
```diff
@@ -19,17 +19,20 @@ class Coupling(BaseModel):
...
         str | None
     ) = None  # the well package that will be used for coupling when sprinkling is active
-    mf6_msw_node_map: FilePath  # the path to the node map file
-    mf6_msw_recharge_map: FilePath  # the path to the recharge map file
-    mf6_msw_sprinkling_map: FilePath | None = (
-        None  # the path to the sprinkling map file
-    )
+    mf6_msw_node_map: Optional[FilePath] = None  # the path to the node map file
+    mf6_msw_recharge_map: Optional[FilePath] = None  # the pach to the recharge map file
+    mf6_msw_sprinkling_map: Optional[
```
Comment (Hofer-Julian) [inline imod_coupler/drivers/ribametamod/config.py:33]: This has the same behaviour.
Please use the “ | None” syntax, it is the modern way of expressing this.

File: imod_coupler/drivers/ribametamod/config.py [hunk 2]
```diff
@@ -9,8 +9,8 @@
 
 class Kernels(BaseModel):
     modflow6: Modflow6
-    ribasim: Ribasim
-    metaswap: Metaswap
+    ribasim: Optional[Ribasim]
+    metaswap: Optional[Metaswap]
```
Comment (Hofer-Julian) [inline imod_coupler/drivers/ribametamod/config.py:13]:
```suggestion
    ribasim: Ribasim | None = None
    metaswap: Metaswap | None = None
```

If you don't set `=None` the key will still be required

File: imod_coupler/drivers/ribametamod/config.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import os
 from pathlib import Path
-from typing import Any, Dict, List
+from typing import Any, Dict, List, Optional
```
Comment (Hofer-Julian) [inline imod_coupler/drivers/ribametamod/config.py:3]:
```suggestion
from typing import Any, Dict, List
```

File: imod_coupler/drivers/ribametamod/ribametamod.py [hunk 4]
```diff
@@ -213,25 +248,32 @@ def update(self) -> None:
...
-
-        # exchange drainage fluxes from MODFLOW 6 to Ribasim
-        self.exchange_rib2mod()
 
-        # Update Ribasim until current time of MODFLOW 6
-        self.ribasim.update_until(self.get_current_time() * days_to_seconds(self.delt))
+        if self.has_metaswap:
+            self.update_MODFLOW6_MetaSWAP()
+        else:
+            self.update_MODFLOW6()
```
Comment (HendrikKok) [inline imod_coupler/drivers/ribametamod/ribametamod.py:259]: I would use self.mf6.update() directly

File: imod_coupler/drivers/ribametamod/config.py [hunk 5]
```diff
@@ -19,14 +19,15 @@ class Coupling(BaseModel):
...
+    mf6_msw_mappings: Dict[str, str] | None = None
 
-    enable_sprinkling: bool  # true whemn sprinkling is active
-    mf6_msw_recharge_pkg: str  # the recharge package that will be used for coupling
-    mf6_msw_well_pkg: (
-        str | None
-    ) = None  # the well package that will be used for coupling when sprinkling is active
-    mf6_msw_node_map: FilePath  # the path to the node map file
-    mf6_msw_recharge_map: FilePath  # the path to the recharge map file
+    enable_sprinkling: Optional[bool] = False  # true whemn sprinkling is active
```
Comment (Hofer-Julian) [inline imod_coupler/drivers/ribametamod/config.py:24]:
```suggestion
    enable_sprinkling: bool = False  # true when sprinkling is active
```
No need for Optional here

**Review by Hofer-Julian — COMMENTED** [review]
Apart from the comments, the code looks good to me

File: imod_coupler/drivers/ribametamod/ribametamod.py [hunk 6]
```diff
@@ -213,25 +248,32 @@ def update(self) -> None:
...
+        if self.has_ribasim:
+            # exchange drainage fluxes from MODFLOW 6 to Ribasim
+            self.exchange_rib2mod()
+            # Update Ribasim until current time of MODFLOW 6
+            self.ribasim.update_until(
+                self.get_current_time() * days_to_seconds(self.delt)
+            )
 
-    def update_modflow6_metaswap(self) -> None:
+    def update_MODFLOW6_MetaSWAP(self) -> None:
```
Comment (HendrikKok) [inline imod_coupler/drivers/ribametamod/ribametamod.py:269]: Please move it back to lower cases. I changed it based on a comment of Julian, I think you inherited this from my original branch.

**Review by HendrikKok — COMMENTED** [review]
Nice work! I have two comments

**Review by HendrikKok — APPROVED** [review]

**Pull request merged**